packages/fiber/src/index.tsx

```diff
@@ -1,7 +1,14 @@
-export * from './three-types'
-import * as ReactThreeFiber from './three-types'
-export { ReactThreeFiber }
-export type { BaseInstance, LocalState } from './core/renderer'
+import { Node, ThreeElements } from './core/types'
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements extends ThreeElements {
+      primitive: Omit<Node<any>, 'args'>
+    }
+  }
+}
+
+export type { AttachFnType, AttachType, Node, ThreeElements, Catalogue, Instance, InstanceProps } from './core/types'
```

Comment:
I can't figure out how to export `{ Node }` into the `ReactThreeFiber` namespace. If I point the old way to `./core/types`, we can leak internal types, but more critically it will prevent extending my augmenting `ThreeElements`.

Reply:
The name `Node` is unfortunate here as well since it would otherwise be inferred from DOM types.